cmd/earthly/main.go

```diff
@@ -70,6 +70,7 @@ type earthlyApp struct {
 	commandName string
 	cliFlags
 	analyticsMetadata
+	projectTracker *analytics.ProjectTracker
```

Comment:
@alexcb not sure if this is the place you had in mind for this object.
Alternatively, I can  define it as a global var in the analytics package, but it would then need to be exported, right?

Reply:
I think a global in analytics is better to be honest -- because that's the pattern we already have for analytics. And it prevents you from having to pass them everywhere.

Currently `Count` is a global function which just forwards it on to the unexported global `counts` pointer.

```
// Count increases the global count of (subsystem, key) which then gets reported when CollectAnalytics is called.
func Count(subsystem, key string) {
        counts.Count(subsystem, key)
}
```

maybe you can do something similar:

```
func AddProjectFromCommandLine(org, project string) {
    projectTracker.AddProjectFromCommandLine(org, prject)
}
```

normally for regular features, these global (call 'em singletons?) might be frowned upon, but analytics are designed to just collect data from the lifespan of the cli running, and fires them off just at the end.

Reply:
Just for some extra context on singletons being frowned upon: usually the biggest downside of the singleton pattern is that any tests against singletons are going to experience test pollution. The tests usually _must_ be run in serial (it's hard to assert on your test's effects when other tests may or may not be running) and often have to be run in a predictable order. And that often means that `go test -run TestSomething` will fail when `go test` passes, because `TestSomething` relies on the effects of a previous test.

But yeah, analytics are a case where it's mostly expected :shrug:

I think a shorter name works just as well for that function, though. e.g. `AddCLIProject(org, project string)` seems to provide roughly the same amount of information in ~half the characters.